aws-storage-s3/src/main/java/com/amplifyframework/storage/s3/IdentityIdProvider.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amplifyframework.storage.s3;
+
+/**
+ * Interface for providing identity ID of the authorized
+ * requester.
+ */
+public interface IdentityIdProvider {
+    /**
+     * Returns the identity ID.
```

Comment:
We should figure out what an Identity ID actually is (lol), and update the Javadoc ...